app/scripts/views/connect_another_device.js

```diff
@@ -0,0 +1,222 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * If the user verifies their email in an instance of Firefox
+ * that other than the one they used to sign up, suggest
+ * that they sign in. If the user verifies in a non-Firefox
+ * browser, they are nudged to install Firefox for Android or iOS.
+ */
+define(function (require, exports, module) {
+  'use strict';
+
+  const Cocktail = require('cocktail');
+  const Constants = require('lib/constants');
+  const ExperimentMixin = require('views/mixins/experiment-mixin');
+  const FormView = require('views/form');
+  const MarketingMixin = require('views/mixins/marketing-mixin');
+  const Template = require('stache!templates/connect_another_device');
+  const Url = require('lib/url');
+  const UserAgent = require('lib/user-agent');
+
+  const View = FormView.extend({
+    template: Template,
+
+    initialize (options = {}) {
+      this._createView = options.createView;
+    },
+
+    events: {
+      'click #signin': '_onSignInClick'
+    },
+
+    showChildView (ChildView, options) {
+      this.$('.child-view').append('<div>');
+      options.el = this.$('.child-view > div');
+      const childView = this._createView(ChildView, options);
+      return childView.render()
+        .then(() => this.trackChildView(childView));
+    },
+
+    beforeRender () {
+      return this.getAccount().isSignedIn()
+        .then((isSignedIn) => {
+          this.model.set('isSignedIn', isSignedIn);
+
+          this.notifier.trigger(`connectAnotherDevice.signedin.${isSignedIn}`);
+          if (this._canSignIn()) {
+            this.notifier.trigger('connectAnotherDevice.signin.eligible');
+          } else {
+            this.notifier.trigger('connectAnotherDevice.signin.ineligible');
+          }
+        });
+    },
+
+    getAccount () {
+      if (! this.model.get('account')) {
+        this.model.set('account', this.user.initAccount({}));
+      }
+
+      return this.model.get('account');
+    },
+
+    context () {
+      const email = this._getEmail();
+      const signInContext = this._getSignInContext();
+      const escapedSignInUrl = this._getEscapedSignInUrl(signInContext, email);
+      const uap = this._getUap();
+      const isAndroid = uap.isAndroid();
+      const isIos = uap.isIos();
+      const isFirefoxIos = isIos && uap.isFirefox();
+      const isOtherIos = isIos && ! isFirefoxIos;
+      const isOther = ! isAndroid && ! isIos;
```

Comment:
We kind of sort of get this information via the "marketing impression" counts, but I think you are right, we should make this explicit.